expression/builtin_json.go

```diff
@@ -136,6 +154,46 @@ func jsonModify(args []types.Datum, mt json.ModifyType, sc *variable.StatementCo
 	return d, nil
 }
 
+// jsonModify1 is similar with `jsonModify`, but for new framework.
+func jsonModify1(args []Expression, row []types.Datum, mt json.ModifyType, sc *variable.StatementContext) (res json.JSON, isNull bool, err error) {
+	res, isNull, err = args[0].EvalJSON(row, sc)
+	if isNull || err != nil {
+		return res, isNull, errors.Trace(err)
+	}
+	pathExprs := make([]json.PathExpression, 0, (len(args)-1)/2+1)
+	values := make([]json.JSON, 0, (len(args)-1)/2+1)
+	for i, arg := range args[1:] {
```

Comment:
can we separate the logic of evaluating `pathExprs` and `values` to two `for` loops ?